### PR TITLE
Update scrape to extract data

### PR DESCRIPTION
**Pull request by ankhuetang**
update scrape + crime controllers

### CODE REVIEW

File: server/controllers/map-controllers.js [hunk 1]
```diff
@@ -66,11 +66,24 @@ const getSearchByAddress = async (req, res, next) => {
...
+	if (crime.length === 0) {
+		console.log('No crime data available');
+	}
+
+	// 7. Respond Object (2 key: safetyScore, traffic)
 	res.status(201).json({
 		Place: place,
 		SafetyScore: safetyScore,
 		Traffic: traffic,
+		Crime: crime,
```
Comment (khoa165) [inline server/controllers/map-controllers.js:86]: usually attributes should be lowercase, pretty odd naming pattern here

File: server/util/map.js [hunk 2]
```diff
@@ -157,6 +158,54 @@ async function getTrafficByLocation(coordinates) {
...
+			location: {
+				coordinates: [crime.location.longtitude, crime.location.latitude],
+			},
+		});
+
+		try {
+			const sess = await mongoose.startSession();
+			sess.startTransaction();
+			await createdCrime.save({ session: sess });
+			await sess.commitTransaction();
```
Comment (khoa165) [inline server/util/map.js:178]: I wonder why we are using transaction here? I thought that if we do, shouldn't the transaction be outside of for loop. Usually, a transaction is used if we have multiple actions, and we want they all succeed or all fail. Here your session has only one action before you commit

File: server/util/map.js [hunk 3]
```diff
@@ -157,6 +158,54 @@ async function getTrafficByLocation(coordinates) {
...
+			console.error(err);
+		}
+	}
+
+	return createdCrimes;
+}
+
+async function getCrimeByLocation(coordinates) {
+	let crime;
+	await Crime.collection.createIndex({ location: '2dsphere' });
```
Comment (khoa165) [inline server/util/map.js:190]: Hmmm why are we creating index here? Shouldn't index be added to the model/schema file instead?

File: server/util/sms.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+//for each subscription, getTraffic and getCrime
+//if no data available, getTrafficInfo and scrapeCrime.
+
+client.messages
+	.create({
+		body: 'Hello from twilio-node',
+		to: '+14019199381', // Text your number
+		from: '+18777504296', // From a valid Twilio number
+	})
+	.then((message) => console.log(message.sid));
```
Comment (khoa165) [inline server/util/sms.js:18]: This looks odd, why is it not inside a function? If this file gets imported, the code would run immediately

File: server/util/webscraping/scheduleScrape.js [hunk 5]
```diff
@@ -1,26 +1,72 @@
...
+				type: doc[5],
+				address: doc[3].substring(6),
+				date: doc[2],
+				location: {
+					longitude: coordinates.lng,
+					latitude: coordinates.lat,
+				},
+			};
+		}
+	});
```
Comment (khoa165) [inline server/util/webscraping/scheduleScrape.js:50]: Can we make a reusable function somehow? I notice this overlaps with the crimes in Providence for the most part

**Review by khoa165 — COMMENTED** [review]
Left some comments to address em

**Comment by ankhuetang** [comment]
created another pull request after resolving the comments

**Pull request closed without merging**